frontend/src/utils/api.js

```diff
@@ -55,30 +55,58 @@ function getEventAttendees(id: string) {
   return fetch(`${API_URL}/events/${id}/attendees?key=${getKey()}`).then(res => res.json())
 }
 
-function addInterviewSchedule(file: File) {
-  var reader = new FileReader()
-  var scheduleString = ''
-  reader.onload = function(e) {
-    scheduleString = reader.result
-    fetch(`${API_URL}/schedule/upload/?key=${getKey()}`, {
-      method: 'POST',
-      headers: {
-        'content-type': 'application/json'
-      },
-      body: JSON.stringify({ schedule: scheduleString })
-    })
-      .then(res => res.json())
-      .then(success => console.log(success))
-      .catch(error => console.log(error))
-  }
-
-  reader.readAsText(file)
+async function addInterviewerSchedules(file: File) {
+  const scheduleString = await readUploadedFile(file)
+  return fetch(`${API_URL}/schedule/uploadInterviewers/?key=${getKey()}`, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    body: JSON.stringify({ data: scheduleString })
+  }).then(res => res.json())
+}
+
+async function addCandidateSchedules(file: File) {
+  const scheduleString = await readUploadedFile(file)
+  return fetch(`${API_URL}/schedule/uploadCandidates/?key=${getKey()}`, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    body: JSON.stringify({ data: scheduleString })
+  }).then(res => res.json())
+}
+
+const readUploadedFile = inputFile => {
+  const reader = new FileReader()
+
+  return new Promise((resolve, reject) => {
+    reader.onerror = () => {
+      reader.abort()
+      reject(new TypeError('Problem parsing input file.'))
```

Comment:
Maybe there's more detailed error information to pass upstream? This TypeError is not particularly descriptive. Normally I'd just let it go, but I have a sneaky feeling that unexpected issues may come up with uploading and parsing files...

Reply:
Sure - this will `reject` upon failing to do an asynchronous read of the file passed into the server, rather than an error during the parsing o the file itself, so maybe I can say "Problem reading input file." 
What do you think?

Reply:
That sounds good to me 👍🏻